cmd/version.go

```diff
@@ -0,0 +1,44 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cmd
+
+import (
+	"os/exec"
+	"strings"
+)
+
+var (
+	genVersion = "unknown"
+	isDeb      = false
+)
+
+func Version() string {
+	if isDeb {
+		return genVersion
+	}
+
+	// not packaged, so maybe running from the tree?
+	if out, err := exec.Command("./mkversion.sh", "-o").CombinedOutput(); err == nil {
```

Comment:
I don't have an issue with it being unknown if unknown; I have an issue with somebody doing `go generate` once, and then always having the wrong version.